src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -72,7 +87,7 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             if ( from == null || to == null || actual == null)
                 throw new ArgumentException( "Cannot compare using a null reference", "actual" );
```

Comment:
This is a fair comment, although potentially a separate pull request, depending how much further work @JustinRChou wants to do on this. 🙂 I'm just very aware this PR has already grown significantly from the original #1674, and there's been a lot of back and forth - it might be good to get this one merged and look at the above separately!

I'll leave that decision with @JustinRChou 😄